File: Bookmark Digest and Homepage/convertscript.js
const fs = require('fs');
const yaml = require('js-yaml');
const fetch = require('node-fetch');
const { parse } = require('node-html-parser');

// Define the input file containing the list of websites
const inputFile = 'websites.txt';

// Define the output file for the YAML data
const outputFile = 'websites.yaml';

// Read the input file and parse the URLs
const urls = fs.readFileSync(inputFile, 'utf-8').trim().split('\n');

// Define a function to extract the title from a web page
async function getTitle(url) {
  try {
    const res = await fetch(url);
    const html = await res.text();
    const root = parse(html);
    const titleElement = root.querySelector('title');
    return titleElement ? titleElement.text.trim() : null;
  } catch (error) {
    console.error(`Error fetching title for ${url}: ${error}`);
    return null;
  }
}

// Define the YAML data structure
const data = [];

// Loop through the URLs and add them to the data structure
for (const url of urls) {
  const title = await getTitle(url) || url;
  data.push({ url, title });
}

// Write the YAML data to the output file
fs.writeFileSync(outputFile, yaml.safeDump(data));

File: Bookmark Digest and Homepage/convertscript.py
import os
import yaml
import requests
from bs4 import BeautifulSoup

# Define the input file containing the list of websites
input_file = 'websites.txt'

# Define the output file for the YAML data
output_file = 'websites.yaml'

# Read the input file and parse the URLs
with open(input_file, 'r') as f:
    urls = [line.strip() for line in f]

# Define a function to extract the title from a web page
def get_title(url):
    try:
        res = requests.get(url)
        soup = BeautifulSoup(res.text, 'html.parser')
        title = soup.title.string.strip() if soup.title else url
        return title
    except Exception as e:
        print("Error fetching title for {}: {}".format(url, e))
        return url

# Define the YAML data structure
data = []

# Loop through the URLs and add them to the data structure
for url in urls:
    title = get_title(url)
    data.append({'url': url, 'title': title})

# Append the YAML data to the output file
if os.path.exists(output_file):
    with open(output_file, 'r') as f:
        existing_data = yaml.safe_load(f) or []
    data = existing_data + data

with open(output_file, 'w') as f:
    yaml.dump(data, f)
